docs/ipfs-cluster-guide.md

```diff
@@ -118,18 +186,19 @@ In normal operation, all three states are in sync, as updates to the *shared sta
+We call a static cluster, that in which the set of `cluster/peers` is fixed, where "peer add"/"peer rm"/bootstrapping operations don't happen (at least normally) and where every cluster member is expected to be running all the time.
 
 Static clusters are a way to run ipfs-cluster in a stable fashion, since the membership of the consensus remains unchanged, they don't suffer the dangers of dynamic peer sets, where it is important that operations modifying the peer set suceed for every cluster member.
 
 Static clusters expect every member peer to be up and responding. Otherwise, the Leader will detect missing heartbeats start logging errors. When a peer is not responding, ipfs-cluster will detect that a peer is down and re-allocate any content pinned by that peer to other peers. ipfs-cluster will still work as long as there is a Leader (half of the peers are still running). In the case of a network split, or if a majority of nodes is down, cluster will be unable to commit any operations the the log and thus, it's functionality will be limited to read operations.
 
 ## Dynamic cluster membership considerations
 
-We call a dynamic cluster, that in which the set of `cluster_peers` changes. Nodes are bootstrapped to existing cluster peers, the "peer add" and "peer rm" operations are used and/or the `leave_on_shutdown` configuration option is enabled. This option allows a node to abandon the consensus membership when shutting down. Thus reducing the cluster size by one.
+We call a dynamic cluster, that in which the set of `cluster/peers` changes. Nodes are bootstrapped to existing cluster peers, the "peer add" and "peer rm" operations are used and/or the `cluster/leave_on_shutdown` configuration option is enabled. This option allows a node to abandon the consensus membership when shutting down. Thus reducing the cluster size by one.
 
 Dynamic clusters allow greater flexibility at the cost of stablity. Join and leave operations are tricky as they change the consensus membership and they are likely to create bad situations in unhealthy clusters. Also, bear in mind than removing a peer from the cluster will trigger a re-allocation of the pins that were associated to it. If the replication factor was 1, it is recommended to keep the ipfs daemon running so the content can actually be copied out to a daemon managed by a different peer.
 
+Peers joining an existing cluster should have a non-divergent state. That means: their consensus `ipfs-cluster-data` folder should be empty or, if not, it should contain data belonging to the existing cluster. A joining peer should have not been running on it's own separately from the running cluster. When a peer leaves or is removed, any existing peers will be saved as `bootstrap` peers, so that it is not easy to start the departing peer in "single-peer-mode" by mistake, as that would essentially create a diverging state, preventing it to re-join its previous cluster cleanly.
```

Comment:
Typo:  should be "on its own" (no apostrophe).  Thanks for making our bootstrap save on removal explicit

Reply:
Fixed. Thanks for finding all these typos!